File: 2020/cf537B3.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    long long int n,m,k,sm=0;
    cin>>n>>k>>m;

    long long int arr[n];
    for(long long int i=0;i<n;i++){
        cin>>arr[i];
        sm=sm+arr[i];
    }
    sort(arr,arr+n);
    long long int i=0;

    long long int ctr=n;
    while(i<n){
        if(i==n-1){
            if(m>0){
                sm=sm+min(m,k);
                m=m-min(m,k);
            }
        }else{
            if(m>0){
                if((double)(sm-arr[i])/(ctr-1) > (double)(sm+min(m,k))/(ctr)){
                    m--;
                    sm=sm-arr[i];
                    ctr--;
                }else{
                    sm=sm+min(m,k);
                    m=m-min(m,k);
                }
            }
        }
        i++;
    }
    cout<<fixed<<setprecision(20)<<(double)sm/ctr;
    return 0;
}

File: 2020/cf537B3.py
x=input().split(' ')

n=int(x[0])
k=int(x[1])
m=int(x[2])

y=input().split(' ')
arr=[]
i=0
while i<n:
    arr.append(int(y[i]))
    i=i+1

arr.sort()
i=min(m,n-1)
m=m-i
sm=0
ctr=0
while i<n:
    if m>0:
        arr[i]=arr[i]+min(m,k)
        m=m-min(m,k)
    sm=sm+arr[i]
    ctr=ctr+1
    i=i+1

print(float(sm)/ctr)
